test/desdemona/functions/sample_functions_test.clj

```diff
@@ -1,13 +1,45 @@
 (ns desdemona.functions.sample-functions-test
-  (:require [clojure.test :refer [deftest is]]
-            [desdemona.functions.sample-functions :refer [transform-segment-shape prepare-rows]]))
+  (:require [clojure.test :refer [deftest is are]]
+            [clojure.java.io :as io]
+            [byte-streams :as bs]
+            [cheshire.core :as json]
+            [desdemona.utils :refer [kwify-map]]
+            [desdemona.functions.sample-functions :refer [prepare-rows message-origin add-message-origin build-row add-original-wrapper]]))
 
-(deftest transform-segment-shape-test
-  (let [got (transform-segment-shape {"line" [:line]} {:line "this is a log line"})
-        expected {"line" "this is a log line"}]
-    (is (= got expected))))
+(deftest add-original-wrapper-test
+  (let [segment {:some :values}
+        got (add-original-wrapper segment)]
+    (is (= got {:original {:some :values}}))))
 
 (deftest prepare-rows-test
   (let [got (prepare-rows {"line" "this is a log line"})
         expected {:rows [{"line" "this is a log line"}]}]
     (is (= got expected))))
+
+(defn raw-example
+  [kind]
+  (kwify-map (json/parse-stream (bs/to-reader (io/file (io/resource (str "test/example_" (name kind) ".json")))) true)))
+
+(defn example
+  [kind]
+  {:original (raw-example kind)})
+
+(deftest message-origin-test
+  (are [expected got] (= expected got)
+    :syslog (message-origin (raw-example :syslog))
+    :json (message-origin (raw-example :json))
+    :falconhose (message-origin (raw-example :falconhose))
+    :cloudpassage (message-origin (raw-example :cloudpassage))))
```

Comment:
This is already a bit better, but you can refactor it more, e.g.:

```
(deftest message-origin-test
  (are [origin] (= origin (message-origin (raw-example origin)))
    :syslog
    :json
    :falconhose
    :cloudpassage))
```

... or even with `doall`.